src/test/java/core/basesyntax/MatrixOperationTest.java

```diff
@@ -1,8 +1,81 @@
 package core.basesyntax;
 
-/**
- * Feel free to remove this class and create your own.
- */
-public class HelloWorldTest {
+import org.junit.Assert;
+import org.junit.Test;
 
-}
\ No newline at end of file
+import java.util.Arrays;
+
+public class MatrixOperationTest {
+    private static final int[][] FIRST_TEST_MATRIX = {
+            {2, 31, 6, 7},
+            {15, 42, 1, 0},
+            {9, 7, 12, 19},
+            {55, 34, 1, 10}
+    };
+    private static final int[][] SECOND_TEST_MATRIX = {
+            {-2, 281, 0},
+            {-15, 0, 1},
+            {3, 7, -12},
+    };
+    private static final int[][] THIRD_TEST_MATRIX = {
+            {-5}
+    };
+
+    private static final int[] FIRST_EXPECTED_DIAGONAL = {2, 42, 12, 10};
+    private static final int[] SECOND_EXPECTED_DIAGONAL = {-2, 0, -12};
+    private static final int[] THIRD_EXPECTED_DIAGONAL = {-5};
+
+    @Test
+    public void getDiagonal() {
```

Comment:
You need to break down method behavior into different scenarios (if they exist). And write test case per scenario. We should not be testing different scenarios in one test case.